lib/galaxy/tools/__init__.py

```diff
@@ -1802,17 +1805,19 @@ def to_json(self, trans, kwd={}, job=None, workflow_building_mode=False):
         """
         history_id = kwd.get('history_id', None)
         history = None
-        try:
-            if history_id is not None:
-                history = self.history_manager.get_owned(trans.security.decode_id(history_id), trans.user, current_history=trans.history)
-            else:
-                history = trans.get_history()
-            if history is None and job is not None:
-                history = self.history_manager.get_owned(job.history.id, trans.user, current_history=trans.history)
-            if history is None:
-                raise exceptions.MessageException('History unavailable. Please specify a valid history id')
-        except Exception as e:
-            raise exceptions.MessageException('[history_id=%s] Failed to retrieve history. %s.' % (history_id, str(e)))
+        if not workflow_building_mode or workflow_building_mode is workflow_building_modes.USE_HISTORY:
```

Comment:
I'd be more explicit (i.e. use less magic) with:
```python
        if workflow_building_mode in (workflow_building_modes.DISABLED, workflow_building_modes.USE_HISTORY):
```

Reply:
True, that looks better!

Reply:
Actually that wouldn't work, because in checks equality, right ?
workflow_building_modes are
```
workflow_building_modes = Bunch(DISABLED=False, ENABLED=True, USE_HISTORY=1)
```
So something like:
```
workflow_building_modes.ENABLED in (workflow_building_modes.DISABLED, workflow_building_modes.USE_HISTORY)
```
would translate to `True in (False, 1)`, which returns True, and that isn't correct here.

Reply:
But this is prone to confusion -- I didn't know workflow_building_mode could be False, 1 or True before this PR and I spent a moment figuring this out. Maybe we could change `workflow_building_modes` to Bunch(DISABLED='disabled', ENABLED='enabled', USE_HISTORY='use_history') ?

Reply:
@nsoranzo what do you think about 2faa56eefe7a8b029ed62b9b4402894c2a26248f

Reply:
@mvdbeek I was going to suggest exactly this "workaround"! Sorry for the wrong inital suggestion.